backend/src/main/java/codezap/auth/provider/basic/BasicAuthCredentialProvider.java

```diff
@@ -0,0 +1,41 @@
+package codezap.auth.provider.basic;
+
+import java.nio.charset.StandardCharsets;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import codezap.auth.provider.CredentialProvider;
+import codezap.global.exception.CodeZapException;
+import codezap.member.domain.Member;
+import codezap.member.repository.MemberRepository;
+
+@Component
+public class BasicAuthCredentialProvider implements CredentialProvider {
+
+    private final MemberRepository memberRepository;
+
+    public BasicAuthCredentialProvider(MemberRepository memberRepository) {
+        this.memberRepository = memberRepository;
+    }
+
+    @Override
+    public String createCredential(Member member) {
+        return HttpHeaders.encodeBasicAuth(member.getLoginId(), member.getPassword(), StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public Member extractMember(String credential) {
+        String[] loginIdAndPassword = BasicAuthDecoder.decodeBasicAuth(credential);
+        Member member = memberRepository.fetchByLoginId( loginIdAndPassword[0]);
+        checkMatchPassword(member,  loginIdAndPassword[1]);
```

Comment:
```suggestion
        Member member = memberRepository.fetchByLoginId(loginIdAndPassword[0]);
        checkMatchPassword(member, loginIdAndPassword[1]);
```
아주 사소한 수정 제안드려요!
아니면 의도한 공백이었을까요?

Reply:
꼼꼼 짱수
꼼장 수정할게용